packages/amplify-util-mock/src/api/api.ts

```diff
@@ -307,6 +391,71 @@ export class APITest {
     return dynamoEmulator.getClient(this.ddbEmulator);
   }
 
+  private async startOpensearchLocalServer(context: $TSContext, isLocalDBEmpty: boolean) {
+    try {
+      const mockConfig = await getMockConfig(context);
+      const mockSearchableResourceDirectory = await this.createMockSearchableArtifacts(context);
+      this.opensearchEmulator = await opensearchEmulator.launch(
+        mockSearchableResourceDirectory, 
+        {
+          port: null, // let the emulator choose the default
+          ...mockConfig,
+        }
+      );
+      if (isLocalDBEmpty) {
+        await this.clearAllIndices(this.opensearchEmulator.url);
+      }
+      return this.opensearchEmulator.url;
+    } catch (error) {
+      throw new AmplifyFault('MockProcessFault', {
+        message: 'Unable to start the local OpenSearch Instance.',
+        link: AMPLIFY_SUPPORT_DOCS.CLI_GRAPHQL_TROUBLESHOOTING.url
+      });
+    }
+  }
+
+  private async clearAllIndices(openSearchURL:URL) {
+    if (openSearchURL) {
+      const errMessage = 'Unable to Clear the local OpenSearch Indices.';
+      try {
+        const url = openSearchURL.toString() + '*';
+
+        const result = await fetch(url, {
+            method: 'DELETE',
+            headers: {
+                'Content-type': 'application/json',
+            }
+        });
+        const status = await result.json();
+        if (!(status?.acknowledged)) {
+          throw new AmplifyFault('MockProcessFault', {
+            message: errMessage,
+            link: AMPLIFY_SUPPORT_DOCS.CLI_GRAPHQL_TROUBLESHOOTING.url
+          });
```

Comment:
What's the purpose of initializing this exception? The outer catch doesn't use it

Reply:
As Ed suggested above, for the Exceptions thrown from inside a `catch`, I've used the `details` part of the `AmplifyFault` to pass this on.